src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -1048,41 +1086,53 @@ private void PopulateParamsFromRestRequest(Dictionary<string, object?> parameter
             }
         }
 
-        /// <summary>
-        /// Authorization check on mutation fields provided in a GraphQL Mutation request.
-        /// </summary>
-        /// <param name="context"></param>
-        /// <param name="parameters"></param>
-        /// <param name="entityName"></param>
-        /// <param name="mutationOperation"></param>
-        /// <exception cref="DataApiBuilderException"></exception>
+        /// <inheritdoc/>
         public void AuthorizeMutationFields(
+            string inputArgumentName,
             IMiddlewareContext context,
+            string clientRole,
             IDictionary<string, object?> parameters,
             string entityName,
             EntityActionOperation mutationOperation)
         {
-            string role = GetRoleOfGraphQLRequest(context);
-
-            List<string> inputArgumentKeys;
-            if (mutationOperation != EntityActionOperation.Delete)
+            if (mutationOperation is EntityActionOperation.Create)
             {
-                inputArgumentKeys = BaseSqlQueryStructure.GetSubArgumentNamesFromGQLMutArguments(MutationBuilder.ITEM_INPUT_ARGUMENT_NAME, parameters);
+                AuthorizeEntityAndFieldsForMutation(context, clientRole, entityName, mutationOperation, inputArgumentName, parameters);
```

Comment:
This method's name: "AuthorizeMutationFields" implies that Entity authorization already occurred. Seems odd that now you are expanding the scope of this method to now call `AuthorizeEntityAndFieldsForMutation(...)`

Reply:
Renaming the method `AuthorizeMutationFields` -> `AuthorizeMutation`.